docs/modules/ROOT/pages/reference/provider-mechanics.adoc

```diff
@@ -0,0 +1,54 @@
+= Crossplane Provider Mechanics
```

Comment:
I would put the page under `explanation`, not reference.

Besides, the nav entry is missing.